src/mame/neogeo/neogeo.cpp

```diff
@@ -1024,9 +1019,11 @@ uint16_t neogeo_base_state::unmapped_r(address_space &space)
 		ret = 0xffff;
 	else
 	{
-		m_recurse = true;
+		if (!machine().side_effects_disabled())
+			m_recurse = true;
 		ret = space.read_word(m_maincpu->pc());
-		m_recurse = false;
+		if (!machine().side_effects_disabled())
+			m_recurse = false;
```

Comment:
`m_recurse` needs to be set during the nested read to avoid the possibility of infinite recursion – it isn’t a side effect per se, and you can’t safely suppress it.